gammapy/estimators/utils.py

```diff
@@ -461,3 +462,131 @@ def compute_lightcurve_doublingtime(lightcurve, flux_quantity="flux"):
     )
 
     return table
+
+
+def get_edges_fixed_bins(fluxpoint, value, axis_name="energy"):
+    """Rebin the flux point to combine value adjacent bins
+
+    Parameters
+    ----------
+    fluxpoint : `FluxPoints`
+        The flux point to rebin
+    value : int
+        Number of bins to combine
+
+    Returns:
+    -------
+    edges_min : `Quantity` or `Time`
+    edges_max : `Quantity` or `Time`
+        Low and High edges of the new axis
+    """
+
+    ax = fluxpoint.geom.axes[axis_name]
+    nbin = ax.nbin
+    if not isinstance(value, int):
+        raise ValueError("Only integer number of bins can be combined")
+    idx = np.arange(0, nbin, value)
+    if idx[-1] < nbin:
+        idx = np.append(idx, nbin)
+    edges_min = ax.edges_min[idx[:-1]]
+    edges_max = ax.edges_max[idx[1:] - 1]
+    return edges_min, edges_max
+
+
+def get_edges_min_sqrt_ts(fluxpoint, value, axis_name="energy"):
+    """Rebin the flux point to combine adjacent bins
+       till a minimum significance is obtained
+
+    Parameters
+    ----------
+    fluxpoint : `FluxPoints`
+            The flux point to rebin
+    value : float
+        The minimum significance desired
+
+    Returns:
+    -------
+    edges_min : `Quantity` or `Time`
+    edges_max : `Quantity` or `Time`
+        Low and High edges of the new axis
+    """
+    ax = fluxpoint.geom.axes[axis_name]
+    nbin = ax.nbin
+
+    e_min, e_max = ax.edges_min[0], ax.edges_max[0]
+    edges_min = np.zeros(nbin) * e_min.unit
+    edges_max = np.zeros(nbin) * e_max.unit
+    i, i1 = 0, 0
+    while e_max < ax.edges_max[-1]:
+        ts = fluxpoint.ts.data[i]
+        e_min = ax.edges_min[i]
+        while ts < value**2 and i < ax.nbin - 1:
+            i = i + 1
+            ts = ts + fluxpoint.ts.data[i]
+        e_max = ax.edges_max[i]
+        i = i + 1
+        edges_min[i1] = e_min
+        edges_max[i1] = e_max
+        i1 = i1 + 1
+    edges_max = edges_max[:i1]
+    edges_min = edges_min[:i1]
+
+    return edges_min, edges_max
+
+
+RESAMPLE_METHODS = {
+    "fixed-bins": get_edges_fixed_bins,
+    "min-sqrt-ts": get_edges_min_sqrt_ts,
+}
```

Comment:
I think your initial implementation offered a third method?

Reply:
Yes, but that is when the user passes fixed bin edges (eg, in the case of recomputing light curves with Bayesian Block edges). In the present scheme, the user can directly create the `MapAxis` for that case add pass it to the resample function

Reply:
Makes sense!